src/main/java/com/sap/cap/cds/intellij/util/NodeJsUtil.java

```diff
@@ -0,0 +1,58 @@
+package com.sap.cap.cds.intellij.util;
+
+import com.intellij.execution.ExecutionException;
+import com.intellij.execution.configurations.GeneralCommandLine;
+import com.intellij.javascript.nodejs.interpreter.local.NodeJsLocalInterpreter;
+import com.intellij.javascript.nodejs.interpreter.local.NodeJsLocalInterpreterManager;
+import org.apache.maven.artifact.versioning.ComparableVersion;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.List;
+import java.util.Optional;
+
+public class NodeJsUtil {
+
+    public static NodeJsLocalInterpreter getInterpreter(ComparableVersion requiredVersion) {
+        Logger.PLUGIN.debug("Searching for Node.js interpreter with required version %s".formatted(requiredVersion));
+
+        for (NodeJsLocalInterpreter interpreter : NodeJsLocalInterpreterManager.getInstance().getInterpreters()) {
+            Optional<ComparableVersion> version = getRuntimeVersion(interpreter.getInterpreterSystemDependentPath());
```

Comment:
I would have assumed that IntelliJ's own manager already provides the version.
It looks like a lot of processes to start before we can start our LSP. On Windows (esp. with virus scanners etc.) this may take a considerable amount of time i.e. delay, i.e. decreased UX.
Is this version information at least cached (persisted) so that the next time we can omit this?

Reply:
`NodeJsLocalInterpreterManager` has an awkward API for that: you can only query the cache and, if absent, have the version determined concurrently and receive it through a `Consumer`. In practice, most versions are *not* already cached upon startup.
Using the `NodeJsLocalInterpreterManager` for that would thus imply that this procedure would first need to be completed for every registered interpreter – whereas with my method, we can break once the first suitable one is found.
Note that the API also isn't documented but we need to rely on decompiled class files.

Reply:
I'll take persistence to another BLI, where we care about persisted settings. Since version determination needs to be done just once per IDE startup as a rule of thumb, I would consider it premature optimization to require persistence right here already – unless you have actually experienced any delay. (The IDE takes several seconds to start up and LSP server is started concurrently.)

Reply:
I see.

>  Note that the API also isn't documented but we need to rely on decompiled class files.

Until the next internal refactoring...

Reply:
> I'll take persistence to another BLI, where we care about persisted settings. Since version determination needs to be done just once per IDE startup as a rule of thumb, I would consider it premature optimization to require persistence right here already – unless you have actually experienced any delay. (The IDE takes several seconds to start up and LSP server is started concurrently.)

I don't understand why the user should be involved in persistence of internal caches. If we check the last modified timestamp plus size of the executable, it should be a good enough cache key.

Reply:
> > Note that the API also isn't documented but we need to rely on decompiled class files.
> 
> Until the next internal refactoring...

Of course they can change it anytime, stability is not guaranteed. But hey, there is no free lunch. :wink:

Reply:
> > I'll take persistence to another BLI, where we care about persisted settings. Since version determination needs to be done just once per IDE startup as a rule of thumb, I would consider it premature optimization to require persistence right here already – unless you have actually experienced any delay. (The IDE takes several seconds to start up and LSP server is started concurrently.)
> 
> I don't understand why the user should be involved in persistence of internal caches. If we check the last modified timestamp plus size of the executable, it should be a good enough cache key.

Of course I don't want the user to define a cache key. That's not what I meant.  :smile: The commonality is the persistence of config settings. (And I would argue this *is* a setting, even if automatically determined.)